src/sentry/api/endpoints/organization_access_request_details.py

```diff
@@ -89,20 +94,29 @@ def get(self, request: Request, organization) -> Response:
                     member__user_is_active=True,
                     member__user_id__isnull=False,
                     team__id__in=request.access.team_ids_with_membership,
-                ).select_related("team")
+                ).select_related("team", "member")
             )
         else:
             # Return empty response if user does not have access
             return Response([])
 
-        return Response(serialize(access_requests, request.user))
+        teams_by_user = OrganizationMember.objects.get_teams_by_user(organization=organization)
+
+        # We omit any requests which are now redundant (i.e. the user joined that team some other way)
+        valid_access_requests: list[OrganizationAccessRequest] = []
+        for access_request in access_requests:
+            if access_request.member.user_id is None:
+                continue
+            if access_request.team_id in teams_by_user[access_request.member.user_id]:
+                continue
+            valid_access_requests.append(access_request)
```

Comment:
Nit: This can be written as a comprehension, like

```suggestion
        valid_access_requests = [
            access_request
            for access_request in access_requests
            if access_request.member.user_id is not None
            and access_request.team_id not in teams_by_user[access_request.member.user_id]
        ]
```